be/src/vec/columns/column_string.cpp

```diff
@@ -391,8 +402,11 @@ ColumnPtr ColumnStr<T>::permute(const IColumn::Permutation& perm, size_t limit)
 template <typename T>
 StringRef ColumnStr<T>::serialize_value_into_arena(size_t n, Arena& arena,
                                                    char const*& begin) const {
-    uint32_t string_size(size_at(n));
-    uint32_t offset(offset_at(n));
+    // https://github.com/apache/doris/pull/10832
+    // Use uint32 instead of size_t to reduce agg key's length.
+    // Size of a line will not exceed 4G, check the size when inserting data into ColumnStr<T>.
+    auto string_size(cast_set<uint32_t>(size_at(n)));
```

Comment:
why here need to check, seem only do check in insert is enough

Reply:
done